src/main/java/carpet/script/annotation/ValueCaster.java

```diff
@@ -0,0 +1,101 @@
+package carpet.script.annotation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import carpet.script.value.AbstractListValue;
+import carpet.script.value.BlockValue;
+import carpet.script.value.BooleanValue;
+import carpet.script.value.EntityValue;
+import carpet.script.value.FormattedTextValue;
+import carpet.script.value.FunctionValue;
+import carpet.script.value.ListValue;
+import carpet.script.value.MapValue;
+import carpet.script.value.NBTSerializableValue;
+import carpet.script.value.NumericValue;
+import carpet.script.value.StringValue;
+import carpet.script.value.ThreadValue;
+import carpet.script.value.Value;
+
+/**
+ * <p>Simple {@link ValueConverter} implementation that casts a {@link Value} into one of its subclasses, either for use directly in parameters or
+ * converters, or as an already working middle step.</p>
+ * 
+ * <p>{@link ValueCaster}s are reused whenever asked for one, since they don't have any complexity.</p>
+ * 
+ * @see #register(Class, String)
+ * 
+ * @param <R> The {@link Value} subclass this {@link ValueCaster} casts to
+ */
+public final class ValueCaster<R> implements ValueConverter<R> // R always extends Value, not explicitly because of type checking
+{
+    private static final Map<Class<? extends Value>, ValueCaster<? extends Value>> byResult = new HashMap<>();
+    static
+    {
+        register(Value.class, "value");
+        register(BlockValue.class, "block");
+        register(EntityValue.class, "entity");
+        register(FormattedTextValue.class, "formatted text");
+        register(FunctionValue.class, "function");
+        register(ListValue.class, "list");
+        register(MapValue.class, "map");
+        register(AbstractListValue.class, "list or similar"); // For LazyListValue basically? Not sure what should use this
```

Comment:
maps and lists are them also. Its for iterators like block iterators

Reply:
Hm... Should the built-in list and map converter check for those instead then (if so, what checks would them have, cause it seems it could cause a bit of chaos in some situations)?

Or should I just keep them in here so functions can decide to receive them manually (aka using `AbstractListValue` as a parameter type)?